### ISSUE TITLE
security: pip-audit reports PYSEC-2024-277 (joblib 1.5.3, disputed) — no fix version exists; allowlist?

### ISSUE THREAD

**Issue by ghostlobster** [ghostlobster/quant-platform#276]
## Summary

`pip-audit -r requirements.txt` (after the existing `--ignore-vuln PYSEC-2022-42969` allowlist) surfaces **one remaining finding**: `joblib@1.5.3` is flagged for `PYSEC-2024-277` / `CVE-2024-34997`. There is **no fix version** — 1.5.3 is the latest joblib release, and the maintainer disputes the finding. Asking how to handle.

## Finding

| Field | Value |
|---|---|
| Package | `joblib` (transitive, via `scikit-learn` / `gensim`) |
| Current installed | `1.5.3` (latest available on PyPI) |
| Advisory IDs | `PYSEC-2024-277` / `CVE-2024-34997` |
| Affected | All versions `>= 0`, including the latest 1.5.3 |
| Fix version | **None** — no patched release exists |
| Severity | High CVSS (deserialization in `NumpyArrayWrapper().read_array()`) |
| Maintainer position | **Disputed** — joblib upstream considers it a false positive: `NumpyArrayWrapper` is only used to deserialize joblib's own trusted on-disk caches, not arbitrary attacker-supplied pickles |

References:
- https://github.com/joblib/joblib/issues/1588 (maintainer "disputed: yes, false positive")
- https://nvd.nist.gov/vuln/detail/CVE-2024-34997

## Current CI behaviour

`.github/workflows/ci.yml:93`:
```yaml
- name: pip-audit
  run: pip-audit -r requirements.txt --ignore-vuln PYSEC-2022-42969 || true
```

The trailing `|| true` masks any pip-audit failure, so this finding does **not** currently break CI. But the SARIF upload still publishes the finding to the Security tab, and the noise makes real findings harder to spot.

## Update classification

**Not a version update** — joblib is already at the latest release. The action is a CI/security-policy update, not a `requirements.txt` bump. Treating this as a **minor config update** (consistent with the existing `PYSEC-2022-42969` allowlist pattern), but flagging here for explicit approval because allowlisting a security finding is a maintenance call, not an automated chore.

## Proposed action (option A — recommended)

Mirror the existing allowlist pattern. In `.github/workflows/ci.yml`:

```diff
- run: pip-audit -r requirements.txt --ignore-vuln PYSEC-2022-42969 || true
+ run: pip-audit -r requirements.txt --ignore-vuln PYSEC-2022-42969 --ignore-vuln PYSEC-2024-277 || true
```

And **drop the `|| true`** so real future findings actually break CI. Once both known-disputed findings are explicitly allowlisted, pip-audit should be allowed to fail-fast on new ones.

Document the rationale in a comment above the step, citing the maintainer dispute and joblib issue #1588.

## Proposed action (option B — stricter)

Drop joblib reliance where possible:
- Audit usage: `grep -r "import joblib\|from joblib" .`
- For sklearn model persistence we already use joblib indirectly via `sklearn.externals.joblib`-style paths or direct `joblib.dump/load` — replacing with `pickle` doesn't change the security calculus (same deserialization risk class), so this is mostly cosmetic.
- Not recommended unless we're prepared to migrate model persistence to a safer format (e.g. ONNX, safetensors-for-sklearn).

## Decision needed

Please confirm:
- [ ] **Option A** — allowlist `PYSEC-2024-277`, drop `|| true`, document the dispute
- [ ] **Option B** — investigate dropping joblib (larger effort)
- [ ] **Defer** — leave as-is until a fixed joblib release appears

Recommendation: **Option A**. Mirrors existing pattern, removes noise, and tightens CI by removing the blanket `|| true`.
